TypeSystem/flow-analysis/reachability_switch_A06_t02.dart

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion switch statement: If `N` is a switch statement of the form
+/// `switch (E) {alternatives}` then:
+/// - Let `before(E) = before(N)`.
+/// - For each `C` in `alternatives` with statement body `S`:
+///   - If `C` is labelled let
+///     `before(S) = conservativeJoin(after(E), assignedIn(N), capturedIn(N))`
+///     otherwise let `before(S) = after(E)`.
+///  - If the cases are exhaustive, then let `after(N) = break(N)` otherwise let
+///    `after(N) = join(after(E), break(N))`.
+///
+/// @description Checks that if a type `T` is made a type of interest in
+/// `alternatives` then it can be promoted in `alternatives` and `after(N)`
```

Comment:
I don't think that's true, neither in the situation where the case is labelled or in the case where it is not. First, `before(S) = after(E)` certainly doesn't introduce anything about types of interest in other cases. Next, in the labelled case, the `conservativeJoin` doesn't transfer information about types of interest anywhere, it only transfers information about being assigned and being write-captured.

In short, an unlabelled case can't see any other cases. A labelled case can see all other cases, but only with respect to 'assigned' and 'captured'.